src/main/exit-app.ts

```diff
@@ -0,0 +1,18 @@
+import { app } from "electron";
+import { WindowManager } from "./window-manager";
+import { appEventBus } from "../common/event-bus";
+import { ClusterManager } from "./cluster-manager";
+import logger from "./logger";
+
+
+export function exitApp() {
+  const windowManager = WindowManager.getInstance<WindowManager>()
+  const clusterManager = ClusterManager.getInstance<ClusterManager>()
+  appEventBus.emit({ name: "service", action: "close" })
+  windowManager?.hide();
+  clusterManager?.stop();
+  logger.info('SERVICE:QUIT');
+  setTimeout(() => {
+    app.exit()
+  }, 1000)
+}
```

Comment:
```suggestion
export function exitApp(timeoutMs = 1000) {
  const windowManager = WindowManager.getInstance<WindowManager>()
  const clusterManager = ClusterManager.getInstance<ClusterManager>()
  appEventBus.emit({ name: "service", action: "close" })
  windowManager?.hide();
  clusterManager?.stop();
  logger.info('SERVICE:QUIT');
  setTimeout(() => {
    app.exit()
  }, timeoutMs)
}
```

Reply:
I'm hesitating to change this as suggested, since the caller requires to know some internal logic of this method to determine proper timeout.

Reply:
Agreed, especially since this is (and should be) only called after everything else